src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/TransactionsViewProjectionHandler.kt

```diff
@@ -0,0 +1,42 @@
+package it.pagopa.ecommerce.eventdispatcher.utils
+
+import it.pagopa.ecommerce.commons.documents.BaseTransactionView
+import it.pagopa.ecommerce.commons.documents.v2.Transaction
+import it.pagopa.ecommerce.commons.domain.v2.TransactionId
+import it.pagopa.ecommerce.eventdispatcher.repositories.TransactionsViewRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.core.env.Environment
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+
+@Component
+object TransactionsViewProjectionHandler {
+
+  lateinit var env: Environment
+
+  // Flag to decide if we have to update the transactions-view collection (true) or not.
+  const val ENV_TRANSACTIONSVIEW_UPDATE_ENABLED_FLAG = "transactionsview.update.enabled"
+
+  @Autowired
+  fun init(environment: Environment) {
+    env = environment
+  }
+
+  fun updateTransactionView(
+    transactionId: TransactionId,
+    transactionsViewRepository: TransactionsViewRepository,
+    viewUpdater: (Transaction) -> Transaction,
+  ): Mono<BaseTransactionView> {
+    val saveEvent = env.getProperty(ENV_TRANSACTIONSVIEW_UPDATE_ENABLED_FLAG, "true").toBoolean()
+    val currentTransactionView =
+      transactionsViewRepository.findByTransactionId(transactionId.value())
+    return if (saveEvent) {
+      currentTransactionView
+        .cast(Transaction::class.java)
+        .map { viewUpdater(it) }
+        .flatMap { transactionsViewRepository.save(it) }
+    } else {
+      currentTransactionView
```

Comment:
maybe this should be the result of applying viewUpdater function, to return updated view also in case of disabled ff. What do you think?